src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2022 Anand Rao, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anand Rao, Rémi Bottinelli
+-/
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import data.set_like.basic
+import category_theory.category.basic
+import category_theory.filtered
+import topology.category.Top.limits
+import category_theory.mittag_leffler
+
+/-!
+# Ends
+
+This file contains a definition of the ends of a simple graph, as sections of the inverse system
+assigning, to each finite set of vertices, the connected components of its complement.
+-/
+
+universes u
+variables {V : Type u} (G : simple_graph V) (K L L' M : set V)
+
+namespace simple_graph
+
+/-- The components outside a given set of vertices `K` -/
+@[reducible] def component_compl := (G.induce Kᶜ).connected_component
+
+variables {G} {K L M}
+
+/-- The connected component of `v` in `G.induce Kᶜ`. -/
+@[reducible] def component_compl_mk (G : simple_graph V) {v : V} (vK : v ∉ K) :
+  G.component_compl K :=
+connected_component_mk (G.induce Kᶜ) ⟨v, vK⟩
```

Comment:
I think you can move `namespace component_compl` up here, and just call this `mk`. (I don't understand why [simple_graph.connected_component_mk](https://leanprover-community.github.io/mathlib_docs/combinatorics/simple_graph/connectivity.html#simple_graph.connected_component_mk) isn't `simple_graph.connected_component.mk`.)

Reply:
I guess it is so that you can write `G.connected_component_mk v` ?
I haven't changed it yet, since I'd expect  @kmill to prefer keeping a similar notation?

Reply:
Good point. LGTM now.

maintainer merge